test/cli/package-import-conflicts/test.out

```diff
@@ -1,16 +1,13 @@
-imports_both_reversed/__package.rb:5: Conflicting import sources for `A::B` https://srb.help/3714
-     5 |  import A
-                 ^
-    imports_both_reversed/__package.rb:4: Conflict from
-     4 |  import A::B
-                 ^^^^
+a/__package.rb:4: Exporting a symbol that isn't owned by this package https://srb.help/3721
+     4 |  export A::B
+          ^^^^^^^^^^^
 
-imports_both/__package.rb:5: Conflicting import sources for `A::B` https://srb.help/3714
-     5 |  import A::B
-                 ^^^^
-    imports_both/__package.rb:4: Conflict from
-     4 |  import A
-                 ^
+ab/ab.rb:3: `A::B` was previously defined as a `class` https://srb.help/4012
+     3 |module A::B
+        ^^^^^^^^^^^
+    a/a.rb:4: Previous definition
+     4 |  class B; end
+          ^^^^^^^
```

Comment:
Is it intentional that we have this "previously defined as a `class`" error now?

Reply:
Yes, this is an effect of everything living in the same namespace again. In file `a/a.rb` the definition looks like this:

```ruby
module A
  class B; end
end
```

While in `ab/ab.rb` it looks like this:

```ruby
module A::B
  class C; end
end
```

This wasn't a problem in the previous design as they lived in separate namespaces, but it's a problem now as we only check visibility in one big namespace.